### PR TITLE
add pause/resume button to bull queues

### PR DESCRIPTION
**Pull request by kwkr**
#### Changes Made
Added `Pause/Resume` button to the queue details view.

#### Potential Risks
I can't think of any.

#### Test Plan
To be discussed

#### Checklist

- [ ] I've increased test coverage
- [x] Since this is a public repository, I've checked I'm not publishing private data in the code, commit comments, or this PR.

### CODE REVIEW

**Comment by kwkr** [comment]
This is only an idea for the feature that would be really convenient to have. So far I'm really missing it and I have to do it using `Bull` library directly. I'm happy to know your opinion on this.

File: src/server/views/helpers/queueHelpers.js [hunk 1]
```diff
@@ -45,7 +45,12 @@ const Helpers = {
     );
     return stats;
   },
-
+  isPaused: async function (queue) {
+    const client = await queue.client;
+    const isPausedValue = await client.get(`bull:${queue.name}:meta-paused`);
+    // @source https://github.com/OptimalBits/bull/issues/1046
+    return isPausedValue == 1;
```
Comment (bradvogel) [inline src/server/views/helpers/queueHelpers.js:52]: use `===`

Comment (bradvogel) [inline src/server/views/helpers/queueHelpers.js:52]: use `===`

File: public/dashboard.js [hunk 2]
```diff
@@ -227,4 +227,58 @@ $(document).ready(() => {
...
+    } else {
+      $(this).prop('disabled', false);
+    }
+  });
+
+  $('.js-resume-queue').on('click', function () {
+    const queueName = $(this).data('queue-name');
+    const queueHost = $(this).data('queue-host');
+
+    const r = window.confirm(
```
Comment (bradvogel) [inline public/dashboard.js:262]: use a more descriptive variable name, eg `const response = `

File: public/dashboard.js [hunk 3]
```diff
@@ -227,4 +227,58 @@ $(document).ready(() => {
         console.error(jqXHR.responseText);
       });
   });
+
+  $('.js-pause-queue').on('click', function () {
+    const queueName = $(this).data('queue-name');
+    const queueHost = $(this).data('queue-host');
+
+    const r = window.confirm(
```
Comment (bradvogel) [inline public/dashboard.js:235]: use a more descriptive variable name, eg `const response = `

File: public/dashboard.js [hunk 4]
```diff
@@ -227,4 +227,58 @@ $(document).ready(() => {
...
+    );
+    if (r) {
+      $.ajax({
+        method: 'PUT',
+        url: `${basePath}/api/queue/${encodeURIComponent(
+          queueHost
+        )}/${encodeURIComponent(queueName)}/pause`,
+      })
+        .done(() => {
+          window.alert(`Queue "${queueHost}/${queueName} paused.`);
```
Comment (bradvogel) [inline public/dashboard.js:246]: Do we need a second alert? The page refreshing and the button changing the 'resume' should be enough user feedback. What do you think?

File: public/dashboard.js [hunk 5]
```diff
@@ -227,4 +227,58 @@ $(document).ready(() => {
...
+    );
+    if (r) {
+      $.ajax({
+        method: 'PUT',
+        url: `${basePath}/api/queue/${encodeURIComponent(
+          queueHost
+        )}/${encodeURIComponent(queueName)}/resume`,
+      })
+        .done(() => {
+          window.alert(`Queue "${queueHost}/${queueName} resumed.`);
```
Comment (bradvogel) [inline public/dashboard.js:273]: same here - do we need the second alert?

**Review by bradvogel — CHANGES_REQUESTED** [review]

**Comment by bradvogel** [comment]
I reviewed this before it might have been ready. Please click Ready for Review when you'd like a second review.

**Comment by bradvogel** [comment]
PR https://github.com/bee-queue/arena/pull/391 included this. Closing

**Pull request closed without merging**